core/src/main/scala/akka/persistence/jdbc/journal/dao/DefaultJournalDao.scala

```diff
@@ -48,21 +52,34 @@ class DefaultJournalDao(
   val queries =
     new JournalQueries(profile, journalConfig.eventJournalTableConfiguration, journalConfig.eventTagTableConfiguration)
 
-  override def delete(persistenceId: String, maxSequenceNr: Long): Future[Unit] = {
-    val actions: DBIOAction[Unit, NoStream, Effect.Write with Effect.Read] = for {
-      _ <- queries.markJournalMessagesAsDeleted(persistenceId, maxSequenceNr)
-      highestMarkedSequenceNr <- highestMarkedSequenceNr(persistenceId)
-      _ <- queries.delete(persistenceId, highestMarkedSequenceNr.getOrElse(0L) - 1)
-    } yield ()
+  override def delete(persistenceId: String, toSequenceNr: Long): Future[Unit] = {
+
+    // note: the passed toSequenceNr will be Long.MaxValue when doing a 'full' journal clean-up
+    // see JournalSpec's test: 'not reset highestSequenceNr after journal cleanup'
```

Comment:
Huh, surprising, I'd expect you often do delete(Long.Max) to drop all events for an entity (and expect to not break projections if re-using the pid) but ok, if the tests say so I guess I'm wrong.